x/cert/keeper/key.go

```diff
@@ -31,18 +30,23 @@ func certificateKey(id types.CertID) []byte {
 }
 
 func certificatePrefix(id sdk.Address) []byte {
-	buf := bytes.NewBuffer(prefixCertificateID)
-	if _, err := buf.Write(id.Bytes()); err != nil {
+	buf := bytes.NewBuffer(types.PrefixCertificateID())
+	if _, err := buf.Write(address.MustLengthPrefix(id.Bytes())); err != nil {
 		panic(err)
 	}
 
 	return buf.Bytes()
 }
 
 func certificateSerialFromKey(key []byte) big.Int {
-	if len(key) < keyAddrPrefixLen+1 {
+	if len(key) < keyAddrPrefixLen {
```

Comment:
I don't think we need this comparison at this time, its redundant with the one below it.

Reply:
Its not redundant, both of them check different indexes before accessing those indexes.
This check exists because we access the index `keyAddrPrefixLen-1` in the next line. If we don't do this check here, then it will anyways panic there with `array index out of range`.

The other check happening below it is also similar, just checking that the index exists before accessing it. If that is not done, then even the next line after that check would panic with `index out of range`.

So, either we should remove both the checks as a panic will anyways be reported (just the panic message will be different), or should keep them both (to have the panic message be as per our choice).

I just kept the convention of having the panic message be of our choice.